src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -84,22 +88,62 @@ public CCDCaseApi(
             .map((caseMapper::from));
     }
 
-    public Claim linkDefendant(String externalId, String defendantId, String authorisation) {
+    /**
+     * LLD https://tools.hmcts.net/confluence/display/ROC/Defendant+linking+with+CCD
+     */
+    public void linkDefendant(String authorisation) {
+        User defendantUser = userService.getUser(authorisation);
+        List<String> letterHolderIds = defendantUser.getUserDetails().getRoles()
+            .stream()
+            .filter(this::isLetterHolderRole)
+            .map(this::extractLetterHolderId)
+            .collect(Collectors.toList());
+
+        if (letterHolderIds.isEmpty()) {
+            return;
+        }
+
         User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
-        CaseDetails caseDetails = getCaseDetailsByExternalId(anonymousCaseWorker, externalId)
-            .orElseThrow(() -> new NotFoundException("Claim not found by external id: " + externalId));
 
+        letterHolderIds
+            .forEach(letterHolderId -> caseAccessApi.findCaseIdsGivenUserIdHasAccessTo(
+                anonymousCaseWorker.getAuthorisation(),
+                authTokenGenerator.generate(),
+                anonymousCaseWorker.getUserDetails().getId(),
+                JURISDICTION_ID,
+                CASE_TYPE_ID,
+                letterHolderId
+            ).forEach(caseId -> linkToCase(defendantUser, anonymousCaseWorker, letterHolderId, caseId)));
+    }
+
+    private void linkToCase(User defendantUser, User anonymousCaseWorker, String letterHolderId, String caseId) {
+        String defendantId = defendantUser.getUserDetails().getId();
+        LOGGER.info("Granting access to case: {} for user: {} with letter-holder id: {}",
+            caseId, defendantId, letterHolderId);
         caseAccessApi.grantAccessToCase(anonymousCaseWorker.getAuthorisation(),
             authTokenGenerator.generate(),
             anonymousCaseWorker.getUserDetails().getId(),
             JURISDICTION_ID,
             CASE_TYPE_ID,
-            caseDetails.getId().toString(),
+            caseId,
             new UserId(defendantId)
         );
 
-        User defendant = userService.getUser(authorisation);
-        return readCase(defendant, caseDetails.getId().toString());
+        LOGGER.info("Revoking access to case: {} for user: {}", caseId, letterHolderId);
+        caseAccessApi.revokeAccessToCase(anonymousCaseWorker.getAuthorisation(),
+            authTokenGenerator.generate(),
+            anonymousCaseWorker.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            caseId,
+            letterHolderId
+        );
+    }
+
+    private boolean isLetterHolderRole(String role) {
+        return role.startsWith("letter")
+            && !role.equals("letter-holder")
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "letter-holder" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1132)